cmd/metal-api/internal/service/integration_test.go

```diff
@@ -94,7 +94,7 @@ func createTestEnvironment(t *testing.T) testEnv {
 	require.NoError(err)
 	go func() {
 		err := grpcServer.Serve()
-		require.Nil(t, err)
+		require.NoError(err)
```

Comment:
Why?

Reply:
because here require was already instatiated with t like this at the beginning of the func 

```
	require := require.New(t)
```

and then `require.Nil(t, err)` checks for a nil t and err is the message if it fails.
I think t was never nil but because this was checked in a goroutine, we never saw this.

Reply:
Removed now all unexpected usage of require and assert